### DISCUSSION TITLE
How to use `pixel_format="luminance_alpha"` for the `hdrfilm` plugin?

### DISCUSSION THREAD

**Discussion by maxfrei750** [mitsuba-renderer/mitsuba3#746] [Q&A] [answered]
I'd like to render grayscale images with an alpha channel, so I set the Mitsuba variant to `cuda_ad_mono` and the `pixel_format` property of the `hdrfilm` to 'luminance_alpha', as per the [docs](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_films.html#high-dynamic-range-film-hdrfilm). When loading the scene, I received the warning:

```
WARN  main  [HDRFilm] Monochrome mode enabled, setting film output pixel format to 'luminance' (was luminance_alpha).
```

Setting the Mitsuba variant to `cuda_ad_rgb` got rid of the warning, but now the rendered images have the shape `(H, W, 4)`, instead of the desired `(H, W, 2)` and, more importantly, my color parameters have three elements, instead of just one.

What am I missing? Thanks!

**Comment by maxfrei750** [comment, marked as answer]
Never mind... Setting the Mitsuba variant to `cuda_ad_rgb` _and_ `pixel_format` to `luminance_alpha`, _does_ produce the desired (H, W, 2) output.

**Comment by njroussel** [comment]
Let me clear up a couple things:

The variant defines the color representation used throughout the entire rendering process. The `pixel_format` choice only defines how the result is then written to the film. This allows us to have an entire RGB simulation but only save the final luminance value of the RGB output in the film.

Using `luminance_alpha` should have been possible with a `mono` variant. I pushed a fix here:https://github.com/mitsuba-renderer/mitsuba3/commit/e3a9a3a3bdecb3c364d604db4fc3a674c057dd6d

**Reply by maxfrei750** [reply]
Thanks for elaborating and the quick fix.
